sklearn/model_selection/_split.py

```diff
@@ -1865,6 +1866,80 @@ def get_n_splits(self, X=None, y=None, groups=None):
         return len(self.unique_folds)
 
 
+class BinnedStratifiedKFold(StratifiedKFold):
+    """Stratified K-Folds cross-validator
+
+    Provides train/test indices to split data in train/test sets.
+
+    This cross-validation object is a variation of KFold that returns
+    stratified folds by binning a regression target. The folds are made
+    by approximately preserving the distribution of targets in each bin.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of folds. Must be at least 2.
+
+    shuffle : boolean, default=False
+        Whether to shuffle each stratification of the data before splitting
+        into batches.
+
+    n_bins : int, default=10
+        How many quantile bins to use.
+
+    random_state : None, int or RandomState
+        When shuffle=True, pseudo-random number generator state used for
+        shuffling. If None, use default numpy RNG for shuffling.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import BinnedStratifiedKFold
+    >>> y = np.arange(11.0)
+    >>> np.random.seed(0)
+    >>> np.random.shuffle(y)
+    >>> X = y + 0.1* np.random.randn(len(y))
+    >>> cv = BinnedStratifiedKFold(n_splits=3)
+    >>> skf = cv.split(X, y)
+    >>> print(cv)
+    BinnedStratifiedKFold(n_bins=5, n_splits=3, random_state=None...)
+    >>> indarr = np.zeros(len(y), dtype=bool)
+    >>> for train_index, test_index in skf:
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [ 5  6  7  8  9 10] TEST: [0 1 2 3 4]
+    TRAIN: [ 0  1  2  3  4  9 10] TEST: [5 6 7 8]
+    TRAIN: [0 1 2 3 4 5 6 7 8] TEST: [ 9 10]
+
+    Notes
+    -----
+    All the folds have size floor(n_samples / n_splits) or
+    floor(n_samples / n_splits) +1,
+    the length is assigned randomly (even if no shuffling is requested)
+    to balance the variance between folds.
+
+    See also
+    --------
+    StratifiedKFold: stratified k-fold generator for classification data
```

Comment:
Mention KFold too, for users looking for other regression-based cross validators